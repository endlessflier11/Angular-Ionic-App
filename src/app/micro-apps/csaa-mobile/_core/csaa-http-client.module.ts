import { NgModule } from '@angular/core';
import { AngularHttpAdapter } from './services/http/adapters/angular-http.adapter';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { IonicHttpAdapter } from './services/http/adapters/ionic-http.adapter';
import { HTTP } from '@ionic-native/http/ngx';
import { UNI_HTTP_ADAPTER } from './services/http/uni-http.model';

@NgModule({
  imports: [HttpClientModule],
  providers: [
    HTTP,
    { provide: UNI_HTTP_ADAPTER, multi: true, useClass: AngularHttpAdapter, deps: [HttpClient] },
    {
      provide: UNI_HTTP_ADAPTER,
      multi: true,
      useClass: IonicHttpAdapter,
      deps: [HTTP],
    },
  ],
  exports: [HttpClientModule],
})
export class CsaaHttpClientModule {}
