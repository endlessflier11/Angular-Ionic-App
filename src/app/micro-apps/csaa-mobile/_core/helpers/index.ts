export * from './router.helpers';
export * from './misc.helpers';
export * from './async.helper';
export * from './error.helper';
export * from './string.helper';
