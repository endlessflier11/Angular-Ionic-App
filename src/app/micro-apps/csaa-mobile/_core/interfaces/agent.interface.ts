export interface Agent {
  name: string;
  phone: string;
  email: string;
  isService: boolean;
}
