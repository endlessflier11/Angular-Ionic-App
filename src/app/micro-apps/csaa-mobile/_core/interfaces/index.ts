export * from './auth.interface';
export * from './router';
export * from './http.interface';
export * from './vehicle.interface';
export * from './ionic-lifecycle.interface';
export * from './analytics.interface';
export * from './policy.interface';
export * from './payment.interface';
export * from './config.interface';
export * from './ngxs-store.interface';
