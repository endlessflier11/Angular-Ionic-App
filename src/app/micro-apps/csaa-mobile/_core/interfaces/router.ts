import { Route, UrlTree } from '@angular/router';
import { NavigationOptions } from '@ionic/angular/providers/nav-controller';
import { Observable } from 'rxjs';

interface AppRoute extends Route {
  name?: string;
  isGroupIndex?: boolean; // If true, it means route is the index of parent. Then it inherits all parent properties
}

export type AppRoutes = AppRoute[];

export interface RouteParam {
  [key: string]: string;
}

export interface NamedRouteConfig {
  name: string;
  path: string;
  children?: NamedRouteConfig[];
}

export interface NamedRoute {
  name: string;
  path: string;
  relativePath: string;
}

export interface CanComponentDeactivate {
  canDeactivate: () =>
    | Observable<boolean | UrlTree>
    | Promise<boolean | UrlTree>
    | boolean
    | UrlTree;
}

export interface CsaaRouteStackItem {
  name: string;
  params?: RouteParam;
  options?: NavigationOptions;
}
