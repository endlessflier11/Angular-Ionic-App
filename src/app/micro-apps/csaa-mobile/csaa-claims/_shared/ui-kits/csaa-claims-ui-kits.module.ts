import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { AgentCardComponent } from './agent-card/agent-card.component';
import { ClaimsHistoryCardComponent } from './claims-history-card/claims-history-card.component';
import { ClaimsInfoCardComponent } from './claims-info-card/claims-info-card.component';
import { WhatToDoCardComponent } from './what-to-do-card/what-to-do-card.component';

const SHARED_COMPONENTS = [
  AgentCardComponent,
  ClaimsHistoryCardComponent,
  ClaimsInfoCardComponent,
  WhatToDoCardComponent,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaClaimsUiKitsModule {}
