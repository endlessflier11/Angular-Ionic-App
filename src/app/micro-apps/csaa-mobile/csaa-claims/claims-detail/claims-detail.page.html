<ion-header [ngClass]="['csaa', currentTheme]">
  <ion-toolbar>
    <ion-buttons slot="start">
      <csaa-back-button (back)="backButtonClick()"></csaa-back-button>
    </ion-buttons>
    <ion-buttons slot="end">
      <ion-button (click)="callClaims()">Call Claims</ion-button>
    </ion-buttons>
  </ion-toolbar>
  <ion-toolbar>
    <ion-title>Claim : {{ (claim$ | async)?.status }}</ion-title>
  </ion-toolbar>
</ion-header>
<ion-content [ngClass]="['csaa', currentTheme]">
  <ion-refresher (ionRefresh)="doRefresh($event.target)" slot="fixed">
    <ion-refresher-content></ion-refresher-content>
  </ion-refresher>
  <csaa-agent-card
    [loading]="loading"
    header="Claims Representative"
    [subHeader]="claimsRepresentativeName"
    [phoneNumber]="claimsPhoneNumber"
    [emailAddress]="claimsEmailAddress"
    [emailSubject]="emailSubjectText"
  >
  </csaa-agent-card>
  <ng-container>
    <csaa-claims-history-card [loading]="loading" [claim]="claim$ | async" [policy]="policy">
    </csaa-claims-history-card>
  </ng-container>
</ion-content>
