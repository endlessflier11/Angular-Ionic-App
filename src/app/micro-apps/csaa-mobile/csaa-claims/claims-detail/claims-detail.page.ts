import { Component, OnDestroy, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { IonRefresher, Platform } from '@ionic/angular';
import { Store } from '@ngxs/store';
import { noop, Observable, Subscription } from 'rxjs';
import { finalize, switchMap } from 'rxjs/operators';
import { CLAIMS_EMAIL, CLAIMS_NUMBER } from '../../constants';
import { withErrorReporter } from '../../_core/helpers';
import { fetchHasFailedFor, unsubscribeIfPresent } from '../../_core/helpers/async.helper';
import { EventType, Policy } from '../../_core/interfaces';
import { Category, EventName } from '../../_core/interfaces/analytics.interface';
import { Claim } from '../../_core/interfaces/claim.interface';
import { AnalyticsService } from '../../_core/services/analytics.service';
import { CallService } from '../../_core/services/call.service';
import { ConfigService } from '../../_core/services/config/config.service';
import { RouterService } from '../../_core/services/router/router.service';
import { SubSink } from '../../_core/shared/subscription.helper';
import { ClaimAction, CustomerAction, PolicyAction } from '../../_core/store/actions';
import { ClaimState } from '../../_core/store/states/claim.state';
import { ContactInfoState } from '../../_core/store/states/contact-info.state';
import { PolicyState } from '../../_core/store/states/policy.state';

@Component({
  selector: 'csaa-claims-detail',
  templateUrl: './claims-detail.page.html',
  styleUrls: ['./claims-detail.page.scss'],
})
export class ClaimsDetailPage implements OnInit, OnDestroy {
  subsink = new SubSink();
  currentTheme: string;
  claimNumber: string;
  claim$: Observable<Claim>;
  policy: Policy;

  emailSubjectText = '';
  claimsRepresentativeName = '';
  claimsEmailAddress: string = CLAIMS_EMAIL;
  claimsPhoneNumber = CLAIMS_NUMBER;

  loading = false;
  private refresher: IonRefresher;
  backButtonSubscription: Subscription;

  constructor(
    private readonly store: Store,
    private readonly callService: CallService,
    private readonly analyticsService: AnalyticsService,
    private readonly platform: Platform,
    private readonly routerService: RouterService,
    private readonly configService: ConfigService,
    private readonly route: ActivatedRoute
  ) {
    this.currentTheme = this.configService.getTheme();
    this.subsink.add(
      fetchHasFailedFor(store, CustomerAction.LoadCustomer, PolicyAction.LoadPolicies).subscribe(
        () => this.routerService.navigateRoot()
      )
    );
  }

  private completeLoading() {
    this.loading = false;
    if (this.refresher) {
      this.refresher.complete();
    }
  }

  ngOnInit() {
    this.route.paramMap.subscribe(
      withErrorReporter((paramMap) => {
        this.claimNumber = paramMap.get('claimNumber');
        this.loadClaimDetails();
      })
    );
  }
  ngOnDestroy() {
    this.subsink.unsubscribe();
  }

  ionViewWillEnter() {
    this.backButtonSubscription = this.platform.backButton.subscribeWithPriority(10, () =>
      this.backButtonClick()
    );
  }

  ionViewWillLeave() {
    unsubscribeIfPresent(this.backButtonSubscription);
  }

  backButtonClick() {
    this.analyticsService.trackEvent(EventName.HOME_ACCESSED, Category.global, {
      event_type: EventType.LINK_ACCESSED,
      link: 'Home',
    });
    this.routerService.back();
  }

  doRefresh(refresher) {
    this.loading = true;
    this.refresher = refresher;
    this.dispatchReloadActions();
  }

  private loadClaimDetails() {
    this.loading = true;
    this.claim$ = this.store.select(ClaimState.claimData(this.claimNumber));
    this.subsink.add(
      this.store.select(ClaimState.claimData(this.claimNumber)).subscribe((claim) => {
        this.claimsRepresentativeName = '';
        this.emailSubjectText = '';
        this.claimsPhoneNumber = CLAIMS_NUMBER;
        this.policy = null;

        if (!!claim) {
          this.policy = this.store.selectSnapshot(PolicyState.policyData(claim.policyNumber));
          this.claimsRepresentativeName = claim.adjuster?.name;
          this.emailSubjectText = `Claim #${claim.number}`;
          this.claimsPhoneNumber =
            claim.adjuster && claim.adjuster.phone
              ? claim.adjuster.phone
              : this.store.selectSnapshot(ContactInfoState.claimsNumber(this.policy.riskState));
        }
      })
    );
    this.dispatchLoadActions();
  }

  callClaims() {
    this.trackContactInitiated('phone');
    this.callService.call(this.claimsPhoneNumber);
  }

  trackContactInitiated(method) {
    this.analyticsService.trackEvent(EventName.CONTACT_INITIATED, Category.claims, {
      event_type: EventType.OPTION_SELECTED,
      selection: 'Contact Claims',
      method: method === 'email' ? 'email' : 'phone',
    });
  }

  private dispatchLoadActions() {
    this.store
      .dispatch(new CustomerAction.LoadCustomer())
      .pipe(
        switchMap(() => this.store.dispatch(new PolicyAction.LoadPolicies())),
        switchMap(() => this.store.dispatch(new ClaimAction.LoadClaims())),
        finalize(() => this.completeLoading())
      )
      .subscribe(withErrorReporter(noop));
  }
  private dispatchReloadActions() {
    this.store
      .dispatch(new CustomerAction.LoadCustomer())
      .pipe(
        switchMap(() => this.store.dispatch(new PolicyAction.LoadPolicies())),
        switchMap(() => this.store.dispatch(new ClaimAction.ReloadClaims())),
        finalize(() => this.completeLoading())
      )
      .subscribe(withErrorReporter(noop));
  }
}
