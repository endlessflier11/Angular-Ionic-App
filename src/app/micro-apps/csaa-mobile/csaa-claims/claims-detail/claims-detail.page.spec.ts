import { Injector } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ActivatedRoute } from '@angular/router';
import { PageTestingModule, StoreTestBuilder } from '@app/testing';
import { IonicModule } from '@ionic/angular';
import { IonicStorageModule } from '@ionic/storage';
import { of } from 'rxjs';
import { Category, EventName } from '../../_core/interfaces';
import { AnalyticsService } from '../../_core/services/analytics.service';
import { CallService } from '../../_core/services/call.service';
import { UiKitsModule } from '../../_core/ui-kits/ui-kits.module';
import { CsaaClaimsUiKitsModule } from '../_shared/ui-kits/csaa-claims-ui-kits.module';

import { ClaimsDetailPage } from './claims-detail.page';
import { CsaaAppInjector } from '../../csaa-app.injector';
import { HttpTestingController } from '@angular/common/http/testing';
import { NgxsModule, Store } from '@ngxs/store';
import { AppState } from '../../../../_core/store';

describe('ClaimsDeatilPage', () => {
  let component: ClaimsDetailPage;
  let fixture: ComponentFixture<ClaimsDetailPage>;
  let analyticService;
  let httpTestingController;
  let store: Store;
  const PARAM_MAP = new Map();
  PARAM_MAP.set('claimNumber', '1003-91-0166');

  beforeEach(() => {
    try {
      TestBed.configureTestingModule({
        declarations: [ClaimsDetailPage],
        providers: [
          { provide: CallService, useValue: { call: jest.fn() } },
          {
            provide: ActivatedRoute,
            useValue: {
              paramMap: of(PARAM_MAP),
            },
          },
        ],
        imports: [
          IonicModule,
          IonicStorageModule.forRoot(),
          UiKitsModule,
          CsaaClaimsUiKitsModule,
          PageTestingModule.withConfig({
            providesPlatform: true,
            providesStorage: true,
            providesAnalytics: true,
            providesConfig: true,
            providesRouter: true,
          }),
          NgxsModule.forRoot([AppState]),
        ],
      }).compileComponents();
      store = TestBed.inject(Store);
      new StoreTestBuilder().withDefaultMocks().resetStateOf(store);

      CsaaAppInjector.injector = TestBed.inject(Injector);
      analyticService = TestBed.inject(AnalyticsService);
      httpTestingController = TestBed.inject(HttpTestingController);
      fixture = TestBed.createComponent(ClaimsDetailPage);
      component = fixture.componentInstance;
      fixture.detectChanges();
    } catch (error) {
      console.error(error);
      fail(error.message);
    }
  });

  afterEach(() => httpTestingController.verify());

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should track home page when clicking on backButton', async () => {
    try {
      fixture.detectChanges();
      await component.backButtonClick();
      expect(analyticService.trackEvent).toHaveBeenCalledWith(
        EventName.HOME_ACCESSED,
        Category.global,
        {
          event_type: 'Link Accessed',
          link: 'Home',
        }
      );
      expect(analyticService.trackEvent).toHaveBeenCalledTimes(1);
    } catch (error) {
      console.error(error.message);
      fail(error.message);
    }
  });
  it('should track call claims', () => {
    component.callClaims();
    expect(analyticService.trackEvent).toHaveBeenCalledWith(
      EventName.CONTACT_INITIATED,
      Category.claims,
      {
        event_type: 'Option Selected',
        method: 'phone',
        selection: 'Contact Claims',
      }
    );
  });

  it('should not fail if state is null', () => {
    expect(component).toBeTruthy();
    const state = store.snapshot();
    state.csaa_app.csaa_customer.csaa_claims = null;
    store.reset(state);
    expect(fixture).toMatchSnapshot();
  });
});
