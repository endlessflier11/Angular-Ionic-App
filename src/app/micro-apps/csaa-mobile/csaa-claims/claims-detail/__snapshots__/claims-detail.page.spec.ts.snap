// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimsDeatilPage should create 1`] = `
<csaa-claims-detail
  __ngContext__={[Function LRootView]}
  analyticsService={[Function AnalyticsMockService]}
  callService={[Function Object]}
  claim$={[Function Observable]}
  claimNumber={[Function String]}
  claimsEmailAddress={[Function String]}
  claimsPhoneNumber={[Function String]}
  claimsRepresentativeName={[Function String]}
  configService={[Function ConfigMockService]}
  currentTheme=""
  emailSubjectText={[Function String]}
  loading="false"
  platform={[Function PlatformMock]}
  policy={[Function Object]}
  route={[Function Object]}
  routerService={[Function RouterMockService]}
  store={[Function Store]}
  subsink={[Function SubSink]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button>
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Back
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
      <ion-buttons
        slot="end"
      >
        <ion-button>
          Call Claims
        </ion-button>
      </ion-buttons>
    </ion-toolbar>
    <ion-toolbar>
      <ion-title>
        Claim : Closed
      </ion-title>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <ion-refresher
      slot="fixed"
    >
      <ion-refresher-content />
    </ion-refresher>
    <csaa-agent-card
      header="Claims Representative"
    >
      <ion-card
        class="csaa-card"
      >
        <ion-card-header>
          <ion-card-title>
             Claims Representative 
          </ion-card-title>
        </ion-card-header>
        <ion-card-content
          class="csaa-item-list"
        >
          <ion-grid>
            <ion-row>
              <ion-col>
                <p
                  class="agent-data"
                >
                  Jacob Dart
                </p>
                <p
                  class="agent-data"
                >
                  216-881-6747
                </p>
              </ion-col>
            </ion-row>
            <ion-row>
              <ion-col
                class="card-buttons-col ion-text-end"
              >
                <ion-button
                  fill="clear"
                >
                  <ion-icon
                    name="call-outline"
                    slot="icon-only"
                  />
                </ion-button>
                <ion-button
                  fill="clear"
                >
                  <ion-icon
                    name="mail-outline"
                    slot="icon-only"
                  />
                </ion-button>
              </ion-col>
            </ion-row>
          </ion-grid>
        </ion-card-content>
      </ion-card>
    </csaa-agent-card>
    <csaa-claims-history-card>
      <ion-card
        class="csaa-card"
      >
        <ion-card-header>
          <ion-card-title>
            <div
              class="display-flex ion-align-items-center"
            >
               CAAC910026006 
            </div>
            <csaa-theme-icon>
              <img
                src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                style="width: 40px; height: 40px;"
              />
            </csaa-theme-icon>
          </ion-card-title>
          <ion-card-subtitle>
             Ford Lgt Convtnl F 
            <br />
             Claim #: 
            <strong>
              1003-91-0166
            </strong>
          </ion-card-subtitle>
        </ion-card-header>
        <ion-card-content
          class="csaa-item-list"
        >
          <ion-list>
            <ion-item>
              <ion-label>
                <p
                  class="date-text"
                  text-small=""
                >
                   Apr 15, 2021 
                </p>
                <h2
                  class="full-bullet"
                >
                   Your Claim was Closed 
                </h2>
                <p
                  text-small=""
                  text-wrap=""
                >
                  If you have any questions, contact your claims representative
                </p>
              </ion-label>
            </ion-item>
            <ion-item>
              <ion-label>
                <p
                  class="date-text"
                  text-small=""
                >
                   Apr 15, 2021 
                </p>
                <h2>
                   Your Claim was Opened 
                </h2>
                <p
                  text-small=""
                  text-wrap=""
                >
                  Cause: Collision Hit and Run
                </p>
              </ion-label>
            </ion-item>
          </ion-list>
        </ion-card-content>
      </ion-card>
    </csaa-claims-history-card>
  </ion-content>
</csaa-claims-detail>
`;

exports[`ClaimsDeatilPage should not fail if state is null 1`] = `
<csaa-claims-detail
  __ngContext__={[Function LRootView]}
  analyticsService={[Function AnalyticsMockService]}
  callService={[Function Object]}
  claim$={[Function Observable]}
  claimNumber={[Function String]}
  claimsEmailAddress={[Function String]}
  claimsPhoneNumber={[Function String]}
  claimsRepresentativeName=""
  configService={[Function ConfigMockService]}
  currentTheme=""
  emailSubjectText=""
  loading="false"
  platform={[Function PlatformMock]}
  policy="null"
  route={[Function Object]}
  routerService={[Function RouterMockService]}
  store={[Function Store]}
  subsink={[Function SubSink]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button>
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Back
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
      <ion-buttons
        slot="end"
      >
        <ion-button>
          Call Claims
        </ion-button>
      </ion-buttons>
    </ion-toolbar>
    <ion-toolbar>
      <ion-title>
        Claim : Closed
      </ion-title>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa"
  >
    <ion-refresher
      slot="fixed"
    >
      <ion-refresher-content />
    </ion-refresher>
    <csaa-agent-card
      header="Claims Representative"
    >
      <ion-card
        class="csaa-card"
      >
        <ion-card-header>
          <ion-card-title>
             Claims Representative 
          </ion-card-title>
        </ion-card-header>
        <ion-card-content
          class="csaa-item-list"
        >
          <ion-grid>
            <ion-row>
              <ion-col>
                <p
                  class="agent-data"
                >
                  Jacob Dart
                </p>
                <p
                  class="agent-data"
                >
                  216-881-6747
                </p>
              </ion-col>
            </ion-row>
            <ion-row>
              <ion-col
                class="card-buttons-col ion-text-end"
              >
                <ion-button
                  fill="clear"
                >
                  <ion-icon
                    name="call-outline"
                    slot="icon-only"
                  />
                </ion-button>
                <ion-button
                  fill="clear"
                >
                  <ion-icon
                    name="mail-outline"
                    slot="icon-only"
                  />
                </ion-button>
              </ion-col>
            </ion-row>
          </ion-grid>
        </ion-card-content>
      </ion-card>
    </csaa-agent-card>
    <csaa-claims-history-card>
      <ion-card
        class="csaa-card"
      >
        <ion-card-header>
          <ion-card-title>
            <div
              class="display-flex ion-align-items-center"
            >
               CAAC910026006 
            </div>
            <csaa-theme-icon>
              <img
                src="/assets/csaa-mobile/vectors/icn-Auto.svg"
                style="width: 40px; height: 40px;"
              />
            </csaa-theme-icon>
          </ion-card-title>
          <ion-card-subtitle>
             Ford Lgt Convtnl F 
            <br />
             Claim #: 
            <strong>
              1003-91-0166
            </strong>
          </ion-card-subtitle>
        </ion-card-header>
        <ion-card-content
          class="csaa-item-list"
        >
          <ion-list>
            <ion-item>
              <ion-label>
                <p
                  class="date-text"
                  text-small=""
                >
                   Apr 15, 2021 
                </p>
                <h2
                  class="full-bullet"
                >
                   Your Claim was Closed 
                </h2>
                <p
                  text-small=""
                  text-wrap=""
                >
                  If you have any questions, contact your claims representative
                </p>
              </ion-label>
            </ion-item>
            <ion-item>
              <ion-label>
                <p
                  class="date-text"
                  text-small=""
                >
                   Apr 15, 2021 
                </p>
                <h2>
                   Your Claim was Opened 
                </h2>
                <p
                  text-small=""
                  text-wrap=""
                >
                  Cause: Collision Hit and Run
                </p>
              </ion-label>
            </ion-item>
          </ion-list>
        </ion-card-content>
      </ion-card>
    </csaa-claims-history-card>
  </ion-content>
</csaa-claims-detail>
`;
